### PR TITLE
Fix/mobile docs header search

### LINKED ISSUE #2172: Keybind for search
image

Shouldn't be too hard to implement I think (but could be a bit OS specific).

Maybe we can also check what's the conventional keybind for that command is. (maybe not Ctrl + K)

This is probably supported by algolia, and probably would appear on version bump.

### LINKED ISSUE #2173: Docs header markup
**Vuestic-ui version:** 1.4.7
 
1. Bug in our docs header markup on mobile

https://user-images.githubusercontent.com/55198465/181690910-336ab7d8-249d-47a1-8525-81253cec6b54.mov

2. We want the search input collapsed by default and expanded (with absolute position and z-index) when focused.

### PR DESCRIPTION
**Pull request by Azgiliat**
## Description

Update docsearch fersion 2 -> 3, now got shortcuts for search. Also search expands on mobile.
When docs is running in dev mode, search results linkns to `localhost` 

Close #2173
Close #2172
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/package.json [hunk 1]
```diff
@@ -19,11 +19,11 @@
     "precommit": "lint-staged --concurrent=false"
   },
   "dependencies": {
+    "@docsearch/js": "3",
```
Comment (asvae) [inline packages/docs/package.json:22]: Can we use "^3.whatever.whatever"? No much difference, just for consistency.

Comment (Azgiliat) [reply packages/docs/package.json:22]: Hm, I'll check it

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue [hunk 2]
```diff
@@ -27,30 +9,22 @@ import { Options, Vue } from 'vue-class-component'
...
+      docsearch({
+        container: '#docsearch',
+        appId: 'DVNV64RN9R',
         indexName: 'vuestic',
-      }, {
-        inputSelector: '#algolia-search-input',
-        debug: true,
-      }))
+        apiKey: 'cd8e70cb466bf6df138543a38c33ea5e',
+        transformItems: process.env.NODE_ENV === 'development'
```
Comment (asvae) [inline packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:21]: Does `yarn serve:docs` work for you? 

I just get an error.

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue [hunk 3]
```diff
@@ -27,30 +9,22 @@ import { Options, Vue } from 'vue-class-component'
...
+          ? (items) => items.map((item) => ({
+            ...item,
+            url: item.url.replace(/^https.*\/en/, `${window.location.origin}/ru`),
```
Comment (asvae) [inline packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:24]: Why would you change language for links?

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue [hunk 4]
```diff
@@ -27,30 +9,22 @@ import { Options, Vue } from 'vue-class-component'
   name: 'Search',
 })
 export default class Search extends Vue {
```
Comment (asvae) [inline packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:11]: * Use composition api/class component syntax.
* File name and component name should be consistent.

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue [hunk 5]
```diff
@@ -27,30 +9,22 @@ import { Options, Vue } from 'vue-class-component'
...
     Promise.all([
-      // @ts-ignore
-      import(/* webpackChunkName: "docsearch" */ 'docsearch.js/dist/cdn/docsearch.min.js'),
-      // @ts-ignore
-      import(/* webpackChunkName: "docsearch" */ 'docsearch.js/dist/cdn/docsearch.min.css'),
-    ]).then(([docsearch]) => {
-      docsearch = docsearch.default
-      docsearch(Object.assign({
-        apiKey: 'be3528055c92da2ea5133b93ed548c6d',
+      import(/* webpackChunkName: "docsearch" */ '@docsearch/js'),
```
Comment (asvae) [inline packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:14]:
```
const {docsearch: default} = await import(/* webpackChunkName: "docsearch" */ '@docsearch/js')
```

would do the same? Looks cleaner to me.

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue [hunk 6]
```diff
@@ -62,152 +36,37 @@ export default class Search extends Vue {
...
-  position: relative;
+#docsearch {
+  flex-grow: 1;
   display: flex;
-  align-items: center;
-  justify-content: space-between;
-  width: 100%;
-  z-index: 1;
-
-  .va-input-wrapper__field {
```
Comment (asvae) [inline packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:75]: image

Can't we make more of a vuestic look? Right now design-wise it doesn't fit with vuestic at all.

We can move this problem into separate task.

File: packages/docs/package.json (hunk 1 above)
Comment (Azgiliat) [reply packages/docs/package.json]: Fixed

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue (hunk 2 above)
Comment (Azgiliat) [reply packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:21]: It was wrong autocoplete, mut be fixed

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue (hunk 5 above)
Comment (Azgiliat) [reply packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:14]: I'm really not sure that we need here dynamic import, so jsut replace it by ordinary static import

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue (hunk 3 above)
Comment (Azgiliat) [reply packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue:24]: Use current locale now

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue (hunk 4 above)
Comment (Azgiliat) [reply packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue]: Fixed

File: packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue (hunk 5 above)
Comment (asvae) [reply packages/docs/src/components/header/components/algolia-search/AlgoliaSearch.vue]: Dynamic import was to speed up initial page load (I think). Basically algolia is included in separate bundle and is sideloaded.

**Review by asvae — APPROVED** [review]
Looks good! 😼

**Pull request merged**